src/actions.js

```diff
@@ -135,50 +139,123 @@ export const clearInitWallet = () => ({ type: types.SET_INIT_WALLET, payload: nu
+ *
+ * wallet {HathorWallet | HathorWalletServiceWallet} wallet object
+ */
+export const fetchHistoryAndBalance = async (wallet) => {
+  // First we get the tokens in the wallet
+  const tokens = await wallet.getTokens();
+
+  const tokensHistory = {};
+  const tokensBalance = {};
+  for (const token of tokens) {
+    /* eslint-disable no-await-in-loop */
+    const balance = await wallet.getBalance(token);
+    const tokenBalance = balance[0].balance;
+    tokensBalance[token] = { available: tokenBalance.unlocked, locked: tokenBalance.locked };
+    const history = await wallet.getTxHistory({ token_id: token });
+    tokensHistory[token] = history.map((element) => mapTokenHistory(element, token));
+    /* eslint-enable no-await-in-loop */
+  }
+
+  return { tokensHistory, tokensBalance };
+};
+
+/**
+ * Fetch pagination history for specific token
+ *
+ * wallet {HathorWallet | HathorWalletServiceWallet} wallet object
+ * token {string} Token uid
+ * history {Array} current token history array
+ */
+export const fetchMoreHistory = async (wallet, token, history) => {
+  const newHistory = await wallet.getTxHistory({ token_id: token, skip: history.length });
+  const newHistoryObjects = newHistory.map((element) => mapTokenHistory(element, token));
+
+  return newHistoryObjects;
+};
+
 export const startWallet = (words, pin) => (dispatch) => {
   // If we've lost redux data, we could not properly stop the wallet object
   // then we don't know if we've cleaned up the wallet data in the storage
   walletUtil.cleanLoadedData();
 
-  const connection = new Connection({
-    network: 'mainnet', // app currently connects only to mainnet
-    servers: ['https://mobile.wallet.hathor.network/v1a/'],
-  });
+  const networkName = 'testnet';
 
-  const beforeReloadCallback = () => {
-    dispatch(activateFetchHistory());
-  };
+  let wallet;
+  // TODO substitute for feature flag when rollout code is implemented
+  if (false) {
+    const connection = new Connection({
+      network: networkName, // app currently connects only to mainnet
+      servers: ['https://node1.foxtrot.testnet.hathor.network/v1a/'],
+    });
 
-  const walletConfig = {
-    seed: words,
-    store: STORE,
-    connection,
-    beforeReloadCallback
-  };
+    const beforeReloadCallback = () => {
+      dispatch(activateFetchHistory());
+    };
 
-  const wallet = new HathorWallet(walletConfig);
+    const walletConfig = {
+      seed: words,
+      store: STORE,
+      connection,
+      beforeReloadCallback
+    };
+
+    wallet = new HathorWallet(walletConfig);
+  } else {
+    const network = new Network(networkName);
+    wallet = new HathorWalletServiceWallet(words, network);
+  }
 
   dispatch(setWallet(wallet));
 
   dispatch(fetchHistoryBegin());
 
+  wallet.on('state', (state) => {
+    if (state === HathorWallet.ERROR) {
+      // ERROR
+      dispatch(fetchHistoryError());
+    } else if (state === HathorWallet.READY || state === 'Ready') {
```

Comment:
Why do we need this `OR`?

Reply:
I agree it feels really weird and should be removed.

Reply:
No need to, we have a `wallet.isReady()` method and I should've used it. Done

Reply:
Seems good to me.